twindb_infrastructure/providers/aws.py

```diff
@@ -25,68 +21,95 @@
-            log.error(err)
-            log.error(cerr)
-            return None
-    except OSError as err:
-        log.error(err)
-        return None
+    :param instance_id: Instance id
+    :type instance_id: str
+    :raise: AwsError
+    :return: object with instance properties
+    :rtype: dict
+    """
+    try:
+        client = boto3.client('ec2')
+        return client.describe_instances(InstanceIds=[instance_id])
+    except ClientError as err:
+        raise AwsError(err)
+    except ValueError as err:
+        raise AwsError('Internal bug in boto3: %s', err)
 
 
 def get_instance_state(instance_id):
-    response = ec2_describe_instance(instance_id)
-
+    """
+    Get state of instance by instance id
+    :param instance_id: Instance id
+    :type instance_id: str
+    :raise: AwsError
+    :return: instance state
+    :rtype str
+    """
     try:
+        response = ec2_describe_instance(instance_id)
         state = response["Reservations"][0]["Instances"][0]["State"]["Name"]
         log.debug("Instance: %s, State: %s" % (instance_id, state))
         return state
-    except ValueError as err:
-        log.error(err)
-        return None
+    except AwsError as err:
+        raise err
 
 
 def get_instance_private_ip(instance_id):
-    response = ec2_describe_instance(instance_id)
-
+    """
+    Get private IP of instance by instance id
+    :param instance_id: Instance id
+    :type instance_id: str
+    :raise: AwsError
+    :return: private ip address of instance
+    :rtype str
+    """
     try:
+        response = ec2_describe_instance(instance_id)
         return response["Reservations"][0]["Instances"][0]["PrivateIpAddress"]
-    except ValueError as err:
-        log.error(err)
-        return None
+    except AwsError as err:
+        raise err
 
 
 def get_instance_public_ip(instance_id):
-    response = ec2_describe_instance(instance_id)
-
+    """
+    Get public IP of instance by instance id
+    :param instance_id: Instance id
+    :type instance_id: str
+    :raise: AwsError
+    :return: public ip address of instance
+    :rtype str
+    """
     try:
+        response = ec2_describe_instance(instance_id)
         return response["Reservations"][0]["Instances"][0]["PublicIpAddress"]
-    except ValueError as err:
-        log.error(err)
-        return None
+    except AwsError as err:
+        raise err
 
 
-def launch_ec2_instance(instance_profile, region=AWS_REGIONS[0],
-                        aws_access_key_id=None, aws_secret_access_key=None,
-                        private_key_file=None):
+def launch_ec2_instance(instance_profile, private_key_file=None):
+    """
+    Launch instance
+    :param instance_profile: Instance profile
+    :param private_key_file: Private key file
+    :type instance_profile: str
+    :type private_key_file: str
+    :return: Instance id
+    :rtype: str
+    """
+    try:
+        client = boto3.client('ec2')
+    except ClientError as err:
+        log.error(err)
```

Comment:
It should raise AwsError here. Also, please add `:raise AwsError:` to the docstring